packages/react/src/components/navigation/IonTabs.tsx

```diff
@@ -144,24 +153,36 @@ export const IonTabs = /*@__PURE__*/ (() =>
         }
       });
 
-      if (!outlet) {
+      if (!outlet && !hasIonTab) {
         throw new Error('IonTabs must contain an IonRouterOutlet');
       }
+      if (outlet && hasIonTab) {
+        throw new Error('IonTabs cannot contain an IonRouterOutlet and an IonTab at the same time');
+      }
       if (!tabBar) {
         throw new Error('IonTabs needs a IonTabBar');
       }
 
+      if (hasIonTab) {
+        return <IonTabsInner {...this.props}></IonTabsInner>;
+      }
+
       return (
         <IonTabsContext.Provider value={this.ionTabContextState}>
           {this.context.hasIonicRouter() ? (
             <PageManager className={className ? `${className}` : ''} routeInfo={this.context.routeInfo} {...props}>
-              <ion-tabs className="ion-tabs" style={hostStyles}>
-                {tabBar.props.slot === 'top' ? tabBar : null}
-                <div style={tabsInner} className="tabs-inner">
-                  {outlet}
-                </div>
-                {tabBar.props.slot === 'bottom' ? tabBar : null}
-              </ion-tabs>
+              <IonTabsInner {...this.props}>
+                {React.Children.map(children, (child: any) => {
```

Comment:
Was there a type issue we ran into that resulted in needing `any` here?

Reply:
It complains with `child.type`. That's most likely the reason that Liam used `any` in a [similar scenario](https://github.com/ionic-team/ionic-framework/blob/f0ba2d7563d7da778843c4e6fe22210187269147/packages/react/src/components/navigation/IonTabs.tsx#L104) in the same file.

Reply:
In my opinion that's not a good reason to disable type checking. Enabling it highlights that a child can be `null` or `undefined` (a case not handled in the implementation) and can also be a `string` or `number`. 

The `type` property comes from expecting that the child is a `React.ReactElement` type. This is one of a few types that could exist. 

The appropriate implementation should be:
```tsx
{React.Children.map(children, (child: React.ReactNode) => {
  if (React.isValidElement(child)) {
    if (child.type === IonTabBar || (child.type as any).isTabBar) {
      /**
       * The modified tabBar needs to be returned to include
       * the context and the overridden methods.
       */
      return tabBar;
    }
  }
  return child;
})}
```

The `as any` could be substituted with a custom type we have for the `isTabBar` type signature. However I think today it is hacked into implementations and doesn't have a type signature.

Reply:
https://github.com/ionic-team/ionic-framework/pull/29772/commits/9c6d2a44772a27d39249aa0d486338a6d4052268